nuheat/util.py

```diff
@@ -68,3 +70,29 @@ def nuheat_to_celsius(nuheat_temperature):
     """
     fahrenheit = nuheat_to_fahrenheit(nuheat_temperature)
     return fahrenheit_to_celsius(fahrenheit)
+
+def get_api_url(brand=config.NUHEAT):
+    brand = brand if brand in config.BRANDS else config.BRANDS[0]
+    return config.API_URL.format(HOSTNAME=config.HOSTNAMES.get(brand))
+
+def get_auth_url(brand=config.NUHEAT):
+    brand = brand if brand in config.BRANDS else config.BRANDS[0]
+    return config.AUTH_URL.format(
+        API_URL=get_api_url(brand=brand)
+    )
+
+def get_thermostat_url(brand=config.NUHEAT):
+    brand = brand if brand in config.BRANDS else config.BRANDS[0]
+    return config.THERMOSTAT_URL.format(
+        API_URL=get_api_url(brand=brand)
+    )
+
+def get_request_headers(brand=config.NUHEAT):
+    brand = brand if brand in config.BRANDS else config.BRANDS[0]
+    return {
+        "Content-Type": "application/x-www-form-urlencoded",
+        "Accept": "application/json",
+        "HOST": config.HOSTNAMES[brand],
+        "DNT": "1",
+        "Origin": get_api_url(brand=brand),
+    }
```

Comment:
just a suggestion... these could be simplified even further if they were properties on the `NuHeat` class.

this would
- save us the trouble of having to check if the `brand` exists in `config.BRANDS` in each method
- simplify the string formatting
- prevent us from needing to include `util` file in the class
- simplify URL lookups as they'd all be tied to the `_session`.

it does pull some strings into the NuHeat class, but i think the simplicity it provides is nice.

e.g.
```
class NuHeat(object):
    ...

    @property
    def _hostname(self):
        return config.HOSTNAMES.get(self.brand)
    
    @property
    def _api_url(self):
        return f"https://{self._hostname}/api"

    @property
    def _auth_url(self):
        return f"{self._api_url}/authenticate/user"

    @property
    def _request_headers(self):
        return {
                "Content-Type": "application/x-www-form-urlencoded",
                "Accept": "application/json",
                "HOST": self._hostname,
                "DNT": "1",
                "Origin": self._api_url,
            }
```

On the thermostat object, we could add the following...

```
class NuHeatThermostat(object):
    ...
    
    @property
    def _url(self):
        return f"{self._session._api_url}/thermostat"
```